src/client/visualizations/line-chart/line-chart.tsx

```diff
@@ -15,718 +15,46 @@
-    const { splits, timezone } = essence;
-
-    if (!splits.length()) return {};
-    const continuousSplit = this.getContinuousSplit();
-    const continuousDimension = this.getContinuousDimension();
-    if (!continuousDimension) return {};
-    const filterRange = this.getFilterRange(essence, continuousSplit, timekeeper);
-    const datasetRange = this.getDatasetXRange(dataset, continuousDimension);
-    const axisRange = union(filterRange, datasetRange);
-    if (!axisRange) return {};
-    const xTicks = this.getLineChartTicks(axisRange, timezone);
-    const scaleX = this.getScaleX(continuousDimension.kind as ContinuousDimensionKind, axisRange, stage.width);
-    return { axisRange, scaleX, xTicks };
-  }
-
-  private getScaleX(kind: ContinuousDimensionKind, { start, end }: PlywoodRange, stageWidth: number): d3.time.Scale<number, number> | d3.scale.Linear<number, number> {
-    const range = [0, stageWidth - VIS_H_PADDING * 2 - Y_AXIS_WIDTH];
-    switch (kind) {
-      case "number": {
-        const domain = [start, end] as [number, number];
-        return d3.scale.linear().domain(domain).range(range);
-      }
-      case "time": {
-        const domain = [start, end] as [Date, Date];
-        return d3.time.scale().domain(domain).range(range);
-      }
-    }
-  }
 
-  private getLineChartTicks(range: PlywoodRange, timezone: Timezone): Array<Date | number> {
-    if (range instanceof TimeRange) {
-      const { start, end } = range;
-      const tickDuration = getBestBucketUnitForRange(range, true) as Duration;
-      return tickDuration.materialize(start, end as Date, timezone);
-    }
-    if (range instanceof NumberRange) {
-      const { start, end } = range;
-      const unit = getBestBucketUnitForRange(range, true) as number;
-      let values: number[] = [];
-      let iter = Math.round((start as number) * unit) / unit;
-
-      while (iter <= end) {
-        values.push(iter);
-        iter += unit;
-      }
-      return values;
-    }
-    throw new Error(`Expected TimeRange or NumberRange. Gor ${range}`);
-  }
-
-  private getFilterRange(essence: Essence, continuousSplit: Split, timekeeper: Timekeeper): PlywoodRange {
-    const maxTime = essence.dataCube.getMaxTime(timekeeper);
-    const continuousDimension = essence.dataCube.getDimension(continuousSplit.reference);
-    const effectiveFilter = essence
-      .getEffectiveFilter(timekeeper);
-    const continuousFilter = effectiveFilter.getClauseForDimension(continuousDimension);
-
-    let range = null;
-    if (continuousFilter instanceof NumberFilterClause) {
-      range = NumberRange.fromJS(continuousFilter.values.first());
-    }
-    if (continuousFilter instanceof FixedTimeFilterClause) {
-      range = TimeRange.fromJS(continuousFilter.values.first());
-    }
-    return this.ensureMaxTime(range, maxTime, continuousSplit, essence.timezone);
+    const scale = calculateXScale(essence, timekeeper, dataset, stage.width - Y_AXIS_WIDTH);
```

Comment:
Responsibility of this component is to calculate xScale (xScale depends exactly on the props on this level so we get lifecycle management for free) and expose fields from `BaseVisualization`